Documentation/ServicingGranularity.md

```diff
@@ -0,0 +1,146 @@
+# Servicing Granularity
+
+This doc is a more detailed description of the problems in [source-build#923
+"Figure out how source-build will work with CoreFX per-package servicing
+policy"](https://github.com/dotnet/source-build/issues/923).
+
+## "Build only if changed"
+
+This is the general rule that if a product hasn't changed, it shouldn't be
+rebuilt and rereleased with a new patch version. It's intuitive and ordinary.
+For example, MSBuild doesn't release new NuGet packages based on downstream
+decisions like a new version of the .NET Core SDK being released.
```

Comment:
Most Linux distributions follow the opposite strategy. Every time a distro package is "released", every component that's part of the source code of the distro package is rebuilt. A component that has source code living in an external distro package is not rebuilt.

Your nuget example is basically:

- Nuget sources are not part of msbuild
- Nuget and msbuild can update/release separately

But the cases we are looking at here **are** part of corefx (or some other repo) sources, but are not "released" from the same branch.

Reply:
It's not intended as a NuGet repo example, "NuGet packages" just meant `.nupkg` here. Will clarify:

> For example, MSBuild doesn't release new `.nupkgs` to the NuGet Gallery based on downstream decisions like a new version of the .NET Core SDK being released.

(I don't understand the MSBuild/NuGet interplay interpretation well enough to understand the contrast vs. CoreFX, maybe clarification is enough?)

And yeah... I'll reduce the scope of "it's ordinary" to  NuGet Gallery packages.

Reply:
When I look at this more closely:

> Every time a distro package is "released", every component that's part of the source code of the distro package is rebuilt. A component that has source code living in an external distro package is not rebuilt.

This is the same strategy, just with the "product" boundary being the distro package rather than a NuGet package. I'm going to try writing an example in each context to compare and hopefully make this clearer. The difference here is that we're treating the whole .NET Core SDK as one distro package and the Microsoft build is chopped up into many NuGet package "products".